eval/compiler/flat_expr_builder.cc

```diff
@@ -991,6 +1404,18 @@ class FlatExprVisitor : public cel::ast_internal::AstVisitor {
     }
   }
 
+  void SetRecursiveStep(std::unique_ptr<DirectExpressionStep> step, int depth) {
+    if (!progress_status_.ok() || PlanningSuppressed()) {
```

Comment:
What happens if planning is suppressed and this call is encountered. I would think that normally we wouldn't get here.